### PR TITLE
Validate Tileset checks for refine property

### PR DESCRIPTION
**Pull request by Rudraksha20**
@lilleyse 
Issue #95: Validate Tileset checks for refine property and its value in root tiles, if refine is defined.

### CODE REVIEW

**Comment by lilleyse** [comment]
Make sure to extend this check to all tiles. The `refine` is only required for `root`, but any other tile may optionally have it.

**Comment by Rudraksha20** [comment]
@lilleyse Made the recommended changes. Resolved merge conflicts and added test for refine is all tilesets.

File: validator/lib/validateTileset.js [hunk 1]
```diff
@@ -36,6 +36,12 @@ function validateTopLevel(tileset) {
         return 'Tileset must declare its geometricError as a top-level property.';
     }
 
+    if (defined(tileset.root.refine)) {
+        if (tileset.root.refine !== 'ADD' && tileset.root.refine !== 'REPLACE') {
+            return 'Refine property in root tileset must have either ADD or REPLACE as its value';
+        }
+    }
```
Comment (lilleyse) [inline validator/lib/validateTileset.js]: This area needs to return an error message if `refine` is not defined for the root.

Also the check that `refine` is `ADD` or `REPLACE` is handled below and doesn't need to be included here.

File: validator/lib/validateTileset.js [hunk 2]
```diff
@@ -102,6 +108,12 @@ function validateTileHierarchy(root, tilesetDirectory) {
             }
         }
 
+        if(defined(tile.refine)) {
```
Comment (lilleyse) [inline validator/lib/validateTileset.js]: Style: `if (`

File: validator/lib/validateTileset.js [hunk 3]
```diff
@@ -102,6 +108,12 @@ function validateTileHierarchy(root, tilesetDirectory) {
...
+            if (tile.refine !== 'ADD' && tile.refine !== 'REPLACE') {
+                return 'Refine property in tileset must have either ADD or REPLACE as its value.';
```
Comment (lilleyse) [inline validator/lib/validateTileset.js]: Change `tileset` to `tile`. Also surround `ADD` and `REPLACE` without quotation marks.

File: validator/specs/lib/validateTilesetSpec.js [hunk 4]
```diff
@@ -67,7 +67,16 @@ describe('validateTileset', function() {
                 expect(message).toBe('Child has geometricError greater than parent');
             }), done).toResolve();
     });
-
+  
+    it('returns error message when refine property of tileset has incorrect value', function(done) {
```
Comment (lilleyse) [inline validator/specs/lib/validateTilesetSpec.js]: Like above, tweak this description.

File: validator/specs/lib/validateTilesetSpec.js [hunk 5]
```diff
@@ -77,6 +86,15 @@ describe('validateTileset', function() {
...
     });
 
+    it('returns error message when refine property of root tile has incorrect value', function(done) {
+        var tileset = clone(sampleTileset, true);
+        tileset.root.refine = 'NEW';
+        expect(validateTileset(tileset)
+            .then(function(message) {
+                expect(message).toBe('Refine property in root tileset must have either ADD or REPLACE as its value');
+            }), done).toResolve();
+    });
```
Comment (lilleyse) [inline validator/specs/lib/validateTilesetSpec.js]: Update this test just to check when `root.refine` is not defined.

**Comment by Rudraksha20** [comment]
@lilleyse 
made the recommended changes. Hope this time I did not miss anything! :)

**Comment by lilleyse** [comment]
The new changes look good!

**Pull request merged**